sycl/include/sycl/ext/intel/esimd/detail/util.hpp

```diff
@@ -209,6 +209,23 @@ auto accessorToPointer(AccessorTy Acc, OffsetTy Offset = 0) {
 }
 #endif // __ESIMD_FORCE_STATELESS_MEM
 
+template <int N, int M, int VStride, int Width, int Stride>
```

Comment:
Thank you. It is still difficult to understand why the constraint is such: `Width == M || ((M / Width) - 1) * VStride + (Width - 1) * Stride < N`. It would help a lot if the pseudo-code would be copied from __esimd_rdregion/wrregion to the commend of these check functions, e.g.;
// This intrinsic computes a vector Result:
//
// \code{.cpp}
// uint16_t EltOffset = Offset / sizeof(T);
// assert(Offset % sizeof(T) == 0);
//
// int NumRows = M / Width;
// assert(M % Width == 0);
//
// int Index = 0;
// for (int i = 0; i < NumRows; ++i) {
//   for (int j = 0; j < Width; ++j) {
//     Result[Index++] = Input[i * VStride +  j * Stride +
//     EltOffset];
//   }
// }
// \endcode

Reply:
Can you do the same for __esimd_wrregion please?
Perhaps, fix the comment in both places too - comment does not say that the write is skipped if mask is 0.

Reply:
Sorry, for confusion. the pseudo-code for __esimd_wrregion is correct. Please just copy it to check function here.
The __esimd_rdregion does not have a mask operand, thus is is all right in the pseudo-code for rdregion.